infrastructure/terraform/gke.tf

```diff
@@ -0,0 +1,56 @@
+provider "google-beta" {
+  # Run 'gcloud auth application-default login' to get credentials.json
+  # credentials = "${file("credentials.json")}"
+  project     = var.project_id
+  region      = var.region
+}
+
+resource "google_container_cluster" "default" {
+  provider = google-beta
+  project  = var.project_id
+  name     = var.gke_cluster_name
+  location = var.region
+
+  network    = google_compute_network.default.name
+  subnetwork = google_compute_subnetwork.default.name
+
+  enable_autopilot = true
+  initial_node_count = var.number_of_nodes
+  node_config {
+    # More info on Spot VMs with GKE https://cloud.google.com/kubernetes-engine/docs/how-to/spot-vms#create_a_cluster_with_enabled
+    # spot = true
+    machine_type = var.machine_type
```

Comment:
I thought autopilot was resource based (we specify the cpu, memory instead of machine type)? Does Terraform not support that?

Reply:
I believe you are right, in theory all we need to specify is name, region and networks and gcp should handle the rest (https://www.hashicorp.com/blog/terraform-adds-support-for-gke-autopilot)

Reply:
Forgot to comment this out, autopilot does provision node pools automatically. This `node_config` was used when I was testing spot instances.